### PR TITLE
Interface for compressible fluid dynamics

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
Very early stage pull request to implement an interface for the compressible fluid solver in FluidDynamicsApplication.

Status:
- [x] Analysis stage
- [x] Parts
- [x] Boundary conditions
- [x] Inital conditions
- [x] Solution
- [x] Results
- [x] Materials
- [x] Added to fluids menu
- [x] Polishing (images, help tooltips, etc)
- [x] Examples
- [x] MainKratos.py

### CODE REVIEW

**Comment by jginternational** [comment]
Hi @EduardGomezEscandell 

Congratulations for your work.

I'm about to merge a big branch, that will change a bit of every app in Kratos.
After the merge, I'll add the changes into your branch, and write you here about the changes, so you con continue with your work and understand the changes.

If you need any help during the app creation process, you can write me or create an issue.

File: kratos.gid/apps/CompressibleFluid/start.tcl [hunk 1]
```diff
@@ -15,13 +15,13 @@ proc ::CompressibleFluid::Init { } {
...
     
-    set app_id Fluid
+    set app_id CompressibleFluid
 
     set kratos_name "FluidDynamicsApplication"
     set dir [apps::getMyDir "CompressibleFluid"]
     set attributes [dict create]
 
-    set prefix COMPF
+    set prefix CF_
```
Comment (rubenzorrilla) [inline kratos.gid/apps/CompressibleFluid/start.tcl:24]: Just mentioning that I'd stick to the current convention (with no underscore).

Comment (EduardGomezEscandell) [reply kratos.gid/apps/CompressibleFluid/start.tcl]: Makes sense. Changed.

**Comment by jginternational** [comment]
Hi @EduardGomezEscandell 

Check the changes I've pushed to this branch. 
I've updated to the new app system, including the app.json file.

Anyway, I'll be tomorrow at the B0 building, so I'll try to help you with this

**Comment by EduardGomezEscandell** [comment]
Great!

I'll take a look-see today and we can talk tomorrow.

**Comment by EduardGomezEscandell** [comment]
There seems to be an issue with the dict->JSON converter. My problem_data is being printed as
```
"problem_data"     : "problem_name shockwave parallel_type OpenMP echo_level 0 start_time 0.0 end_time 7.0]"
```
This matches my input, but is clearly incorrectly formatted. Same thing happens with output processes:
```
"output_processes" : "gid_output {{python_module gid_output_process kratos_module KratosMultiphysics process_name GiDOutputProcess help {This process writes postprocessing files for GiD} Parameters {model_part_name FluidModelPart.fluid_computational_model_part output_name shockwave postprocess_parameters {result_file_configuration {gidpost_flags {GiDPostMode GiD_PostBinary WriteDeformedMeshFlag WriteDeformed WriteConditionsFlag WriteConditions MultiFileFlag SingleFile} file_label time output_control_type time output_interval 1.0 body_output Yes node_output No skin_output No plane_output {} nodal_results {MOMENTUM DENSITY TOTAL_ENERGY PRESSURE TEMPERATURE VELOCITY} gauss_point_results {} nodal_nonhistorical_results {}} point_data_configuration {}}}}} vtk_output {{python_module vtk_output_process kratos_module KratosMultiphysics process_name VtkOutputProcess help {This process writes postprocessing files for Paraview} Parameters {model_part_name FluidModelPart.fluid_computational_model_part output_control_type step output_interval 1 file_format ascii output_precision 7 output_sub_model_parts false output_path vtk_output save_output_files_in_folder true nodal_solution_step_data_variables {MOMENTUM DENSITY TOTAL_ENERGY PRESSURE TEMPERATURE VELOCITY} nodal_data_value_variables {} element_data_value_variables {} condition_data_value_variables {} gauss_point_variables_extrapolated_to_nodes {}}}}]",
```

**Comment by jginternational** [comment]
I'll take a look at it tomorrow. My first recommendation is to remove the `help` parameter

File: kratos.gid/apps/CompressibleFluid/write/writeProjectParameters.tcl [hunk 2]
```diff
@@ -3,10 +3,10 @@ proc ::CompressibleFluid::write::getParametersDict { } {
     set projectParametersDict [dict create]
 
     # Problem data
-    dict set projectParametersDict problem_data [write::GetDefaultProblemDataDict $CompressibleFluid::app_id]
+    dict set projectParametersDict problem_data [write::GetDefaultProblemDataDict [::CompressibleFluid::GetAttribute id]]
```
Comment (EduardGomezEscandell) [inline kratos.gid/apps/CompressibleFluid/write/writeProjectParameters.tcl:6]: Well the issue was here. There was an extra bracket `]` that made the JSON converter think this dict was all a string.

Same thing was happening for output_processes

**Comment by EduardGomezEscandell** [comment]
Closing because I will no longer work on this issue, if anyone whants to take over, be my guest.

**Comment by rubenzorrilla** [comment]
@Marco1410 @jginternational any advance in this regard?

**Comment by jginternational** [comment]
You can test it and merge the Pull request when ready @Marco1410

**Comment by Marco1410** [comment]
Hi @jginternational, I have added a little changes.

Still remains to do:

- [ ] Fix "constrained" parameter of momentum condition: To initial conditions the constrained have to be "false".

- [ ] Fix "null" value of z component for momentum: When I create a new problem this value appear between brackets. This not happen with the examples and I don't now why.

- [ ] For the last, the "time_schemes" appear in GiD to selection them but they don't appear in ProjectParameters file.

The first two items are very important because the problem isn't run successfully without change them.

Captura desde 2023-01-17 15-04-14

I think that with those change the interface would be ready.

**Comment by rubenzorrilla** [comment]
Are all @Marco1410 's bullet points fixed now?

**Review by jginternational — APPROVED** [review]

**Pull request merged**